dataflux_pytorch/benchmark/checkpointing/multinode/fsspecfsdp.py

```diff
@@ -0,0 +1,109 @@
+import os
+
+import gcsfs
+import torch
+
+from pathlib import Path
+from typing import Generator
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.strategies import FSDPStrategy
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+from lightning.pytorch.strategies.fsdp import _METADATA_FILENAME
+from torch.distributed.checkpoint import save, load
+from torch.nn import Module
+from torch.distributed.checkpoint import _fsspec_filesystem as FF
+
+
+class FSSpecFSDPStrategy(FSDPStrategy):
+
+    def __init__(self, path, model, **kwargs):
+        super().__init__(**kwargs)
+        self.model = model
+        self.path = path
+
+    def save_checkpoint(self,
+                        checkpoint,
+                        filepath,
+                        storage_options=None) -> None:
+        if storage_options is not None:
+            raise TypeError(
+                "`FSDPStrategy.save_checkpoint(..., storage_options=...)` is not supported because"
+                " `FSDPStrategy` does not use the `CheckpointIO`.")
+
+        writer = FF.FsspecWriter(self.path, sync_files=False)
```

Comment:
Any specific reason for initializing writer in the save_checkpoint instead of init ?

Reply:
Not really. I had to initialize the reader in `load_checkpoint` because the checkpoint path is different for different for different reads. I moved the initialization of the writer to `save_checkpoint` just so it's consistent.

Reply:
I agree that it would be better to initialize your reader & writer as instance variables inside the init method so you can reuse them instead of recreating them each time save/load is called.

Reply:
I can initialize the writer in init but am lost on how to do the same for reader. The reader needs to be passed the path to the checkpoint that we want to load which is an argument to `FSSpecFSDPStrategy.load_checkpoint` so it's easier to initialize it there. Any tips?

Reply:
Yeah its kind of how it works, and hence we initialize strategy again in every iteration https://github.com/GoogleCloudPlatform/dataflux-pytorch/blob/main/dataflux_pytorch/benchmark/checkpointing/multinode/train.py#L84 

The reason why I am initializing reader/writer outside of save_checkpoint / load_checkpoint is because it is not fair to count the time it takes to initialize as part of save_checkpoint/load_checkpoint